help: Update "private message" -> "direct message".

drrosa opened the pull request:
Updates help center documentation to match new **direct message (DM)** term in Zulip UI.

[CZO discussion](https://chat.zulip.org/#narrow/stream/101-design/topic/.22Private.22.20vs.20.22Direct.22).

**Screenshots and screen captures:**
- https://zulip.com/help/private-messages
<img width="680" alt="image" src="https://user-images.githubusercontent.com/2343554/217730167-9bab8359-a614-4949-9d1b-51cc31ccd6e7.png">

<img width="501" alt="image" src="https://user-images.githubusercontent.com/2343554/217730326-aff191f2-fbf8-4829-a22f-678ed610e731.png">

<img width="501" alt="image" src="https://user-images.githubusercontent.com/2343554/221706643-cc6eb770-e0e7-4274-8216-ec92884a1d51.png">

<img width="669" alt="image" src="https://user-images.githubusercontent.com/2343554/217730458-e26bff80-837b-4006-acd3-452ca37ce7be.png">

- https://zulip.com/help/reading-pms
<img width="572" alt="image" src="https://user-images.githubusercontent.com/2343554/218148448-b23eaea4-f711-4db6-8a75-14cadf9bf2ff.png">

- https://zulip.com/help/restrict-private-messages
<img width="705" alt="image" src="https://user-images.githubusercontent.com/2343554/218148925-39399ded-a116-45d1-9d4f-f1c233a83d9c.png">

- https://zulip.com/help/pm-mention-alert-notifications
<img width="600" alt="image" src="https://user-images.githubusercontent.com/2343554/218149147-904035b9-aae2-4adb-b0d4-22fd2aaae19e.png">


<details>
<summary>Self-review checklist</summary>

- [x] [Self-reviewed](https://zulip.readthedocs.io/en/latest/contributing/code-reviewing.html#how-to-review-code) the changes for clarity and maintainability

Communicate decisions, questions, and potential concerns.

- [x] Highlights technical choices and bugs encountered.
- [x] Calls out remaining decisions and concerns.
- [x] Automated tests.

Individual commits are ready for review (see [commit discipline](https://zulip.readthedocs.io/en/latest/contributing/version-control.html)).

- [x] Each commit is a coherent idea.
- [x] Commit message(s) explain reasoning and motivation for changes.

Completed manual review and testing of the following:

- [x] Visual appearance of the changes.
- [x] Links.
</details>

alya commented:
Thanks, those screenshots look good to me!

@timabbott This is WIP, because not all affected pages have been updated yet.

drrosa commented:
@alya This PR is ready for you.

timabbott commented:
OK. When this is reviewed, we may want to just have @laurynmm append it to the PR doing the renaming, so we can rebase/merge them together as a single action.

drrosa commented:
Hm, there's a failing test, but it doesn't seem related to this PR:
> The Puppeteer frontend tests failed! The failing test was:
    ./tools/test-js-with-puppeteer user-deactivation.test.ts

alya commented:
I think we shouldn't change the mobile instructions, because the tab is still called "private messages" on a tablet, right? In any case, having those instructions have a mix of "private messages" and "direct messages" seems a bit.

Otherwise looks good to me! I have not attempted to verify whether the changes cover everything.

timabbott commented:
Just do a rebase to clear the test failure -- it's a nondeterminstic flake.

timabbott commented on zerver/lib/url_redirects.py:
```diff
@@ -21,6 +21,10 @@ class URLRedirect:
     # Add URL redirects for help center documentation here:
+    URLRedirect("/help/pm-mention-alert-notifications", "/help/dm-mention-alert-notifications"),
+    URLRedirect("/help/restrict-private-messages", "/help/restrict-direct-messages"),
+    URLRedirect("/help/reading-pms", "/help/reading-dms"),
```
@alya do we want to take the opportunity to make this "Reading direct messages" rather than "Reading DMs" (and thus also make the URL not have the abbreviation, here)? Similar question for `/help/dm-mention-alert-notifications`

timabbott commented on help/restrict-direct-messages.md:
```diff
@@ -0,0 +1,38 @@
+
+{!admin-only.md!}
+
+In Zulip, users can exchange direct messages with other users,
```
This can be a follow-up commit, since it's not directly related to the renaming, but should "direct messages" here be a link to the main article on them?

(Or should we have a "See also" on this page).

timabbott commented on help/keyboard-shortcuts.md:
```diff
@@ -81,13 +81,13 @@ in the Zulip app to add more to your repertoire as needed.
+* **Narrow to topic or DM conversation**: <kbd>Shift</kbd> + <kbd>S</kbd>
 
-* **Narrow to all private messages**: <kbd>Shift</kbd> + <kbd>P</kbd>
+* **Narrow to all direct messages**: <kbd>Shift</kbd> + <kbd>P</kbd>
```
@alya just a note on an awkward detail with keyboard shortcuts: the `p/P` keyboard shortcuts are now not super intuitively named, and we already have a `d/D` shortcut. I don't think we have to do anything about this immediately, but wanted to flag it. Probably worth some discussion in the #design thread.

timabbott replied:
Posted https://chat.zulip.org/#narrow/stream/101-design/topic/.22Private.22.20vs.20.22Direct.22/near/1514589 about this.

drrosa commented:
> I think we shouldn't change the mobile instructions, because the tab is still called "private messages" on a tablet, right? In any case, having those instructions have a mix of "private messages" and "direct messages" seems a bit.
> 
> Otherwise looks good to me! I have not attempted to verify whether the changes cover everything.

I thought maybe the mobile app labels would be updated by the time this got merged.
I changed the mobile instructions back to "Private message" and rebased/updated the PR .

alya commented:
Sounds good, thanks! @chrisbobbe @gnprice let's coordinate on documentation changes when we update the mobile UI to rename PMs -> DMs.

alya commented:
@timabbott please take a look.

timabbott commented:
Merged, thanks @drrosa!
